language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java

```diff
@@ -220,6 +222,48 @@ public CompletableFuture<TypesFromSymbolResponse> getTypeFromSymbol(TypeFromSymb
         });
     }
 
+    @JsonRequest
+    public CompletableFuture<TypesFromSymbolResponse> getTypesFromFnDefinition(TypesFromFnDefinitionRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            TypesFromSymbolResponse typeFromSymbolResponse = new TypesFromSymbolResponse();
+            String fileUri = request.getDocumentIdentifier().getUri();
+            Optional<Path> filePath = PathUtil.getPathFromURI(fileUri);
+            if (filePath.isEmpty()) {
+                return typeFromSymbolResponse;
+            }
+            List<ResolvedTypeForSymbol> types = new ArrayList<>();
+            try {
+                Optional<SemanticModel> semanticModel =
+                        this.workspaceManagerProxy.get(fileUri).semanticModel(filePath.get());
+                Optional<Document> document = workspaceManagerProxy.get(fileUri).document(filePath.get());
+                if (semanticModel.isPresent() && document.isPresent()) {
```

Comment:
Shall we `early return` here checking for absence

Reply:
Addressed with 7b7dc194b28a59dbbdc3791f4b3d4f4b1bbfb74e